engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobTimeOutProcessor.java

```diff
@@ -44,32 +45,29 @@ public JobTimeOutProcessor(
 
   @Override
   public void processRecord(final TypedRecord<JobRecord> record) {
-    final long jobKey = record.getKey();
-    final JobState.State state = jobState.getState(jobKey);
+    final var jobKey = record.getKey();
+    final var job = jobState.getJob(jobKey);
+    final var state = jobState.getState(jobKey);
 
-    if (state == State.ACTIVATED) {
-      final JobRecord timedOutJob = record.getValue();
+    final var now = ActorClock.currentTimeMillis();
+    final var deadline = job.getDeadline();
+    final var hasTimedOut = now > deadline;
 
-      stateWriter.appendFollowUpEvent(jobKey, JobIntent.TIMED_OUT, timedOutJob);
-      jobMetrics.jobTimedOut(timedOutJob.getType());
-
-      jobActivationBehavior.publishWork(jobKey, timedOutJob);
+    if (state == State.ACTIVATED && hasTimedOut) {
+      stateWriter.appendFollowUpEvent(jobKey, JobIntent.TIMED_OUT, job);
+      jobMetrics.jobTimedOut(job.getType());
+      jobActivationBehavior.publishWork(jobKey, job);
     } else {
-      final String textState;
-
-      switch (state) {
-        case ACTIVATABLE:
-          textState = "it must be activated first";
-          break;
-        case FAILED:
-          textState = "it is marked as failed";
-          break;
-        default:
-          textState = "no such job was found";
-          break;
-      }
+      final var reason =
+          switch (state) {
+            case ACTIVATED -> "it has not timed out";
```

Comment:
:upside_down_face: A bit inelegant, would be much nicer with switch guards, coming in Java 21.